debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -1286,12 +1280,12 @@ protected String createPatternFromName(String name) {
         for (int i = 0; i < length; i++) {
             char c = name.charAt(i);
 
-            if (likeWildcardCharacters.contains(c) || LIKE_ESCAPE_CHARACTER.equals(c)) {
+            if (likeWildcardCharacters.contains(c) || databaseSearchEscapeCharacter.equals(String.valueOf(c))) {
```

Comment:
Alternatively, before the `for` loop, we can assert that the `databaseSearchEscapeCharacter` string consists of one character and extract the character. Then use the character for comparison.

If some driver uses more than one character for escaping (which I'm not sure is allowed by SQL), the condition `databaseSearchEscapeCharacter.equals(String.valueOf(c))` would have to be reworked as it would never evaluate to `true`.